libc/picolibc: Picolibc fixes

keith-packard opened the pull request:
These are a handful of fixes required to use picolibc as the default C library

keith-packard committed:
samples/tfm_integration: Remove 'noreturn' attribute from main

This conflicts with what the compiler expects for the definition of 'main'
when -ffreestanding is not used.

Signed-off-by: Keith Packard <keithp@keithp.com>

keith-packard committed:
samples/userspace: Set common malloc arena size to zero for syscall_perf

The default malloc arena when using picolibc is too large for the
hifive1_revb target.

Signed-off-by: Keith Packard <keithp@keithp.com>

joerchan reviewed (commented):
Approving TF-M sample change

dcpleung commented on samples/tfm_integration/tfm_psa_test/src/main.c:
```diff
@@ -9,7 +9,6 @@
 void psa_test(void);
 
-__attribute__((noreturn))
 int main(void)
```
Does this need to return something even with the for loop in the end? Some toolchains might complain that there is no return statement.

keith-packard replied:
You have the choice between a warning about a missing return and a warning for unreachable code. It looks like gcc (at least) is happy with the code as-is.

microbuilder replied:
@dcpleung Any further feedback on this one?

dcpleung replied:
No. As long as the compiler is happy, I am fine with this.

keith-packard committed:
samples/cmsis_rtos_v2/timer_synchronization: Set malloc arena size to zero

This test fails when using picolibc with the default malloc arena
value. Fix this by explicitly setting the malloc arena size to zero, as
when using the minimal C library.

Signed-off-by: Keith Packard <keithp@keithp.com>

keith-packard commented:
I moved the modules/audio/sof fix out of this PR as that didn't fit the other changes here which are all minor fixes to samples.
